controllers/secret_controller.go

```diff
@@ -191,6 +207,49 @@ func (r *SecretReconciler) Reconcile(ctx context.Context,
 	}
 }
 
+func (r *SecretReconciler) reconcileTLSSecret(ctx context.Context) error {
+	tlsSecret := &core.Secret{}
+	err := r.client.Get(context.TODO(), kubeTypes.NamespacedName{Name: secrets.TLSSecret,
+		Namespace: r.watchNamespace}, tlsSecret)
+	if err != nil {
+		if k8sapierrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return nil
+		}
```

Comment:
Will this case be hit on a deletion event? If so, we need to reconcile here to remove the old cert and key files from each Windows node

Reply:
Yes this needs cleanup logic which needs to be handled with a finalizer.
Without one the controller will not be able to reliably react to the deletion event before the secret is actually deleted.

Reply:
Or does it need cleanup logic? I suppose all that matters is if a new secret is created the certs on the node are replaced.